ps0.py

```diff
@@ -1,10 +1,94 @@
-# 1. Ask the user to enter a number "x"
-
-
-# 2. Ask the user to enter a number "y"
-
-
-# 3. Prints out the number "x" raised to the power "y"
-
-
-# 4. Prints out the log (base 2) of "x"
\ No newline at end of file
+##FUNCTION 2: LOG FOR ANY BASE
+#Okay so this takes some more explaining. Idk if logs are native to python but numpy and pylab is on page 3 of the instructions, not page 2. In any case, I do know the change of base formula.
+def logb(argument, base):
+    #simple enough, if you just have log base 10
+    return log(argument)/log(base)
+
+##FUNCTION 1: LOG FOR BASE 10
+#The fun part
+#I used logarithmic and exponential equivalence. logb(a) = r == b^r = a
+#It basically solves for r, by granting the expression of 10 to an exponent and comparing it to the argument
+#It's not perfect of course, but it seems to be able to work fine with tested numbers, and adding 0's to the acceptable range makes it more accurate
+def log(argument):
```

Comment:
There is a built-in log function that directly computes log base 2: [math.log2()](https://docs.python.org/3/library/math.html#power-and-logarithmic-functions). But still very impressive effort on creating custom log function.